msal/src/main/java/com/microsoft/identity/client/Prompt.java

```diff
@@ -41,35 +44,47 @@
      * <p>
      * toString override is to enable the correct protocol value of login to be returned instead of "force_login".
      */
-    FORCE_LOGIN,
+    LOGIN,
 
     /**
      * acquireToken will send prompt=consent to the authorize endpoint.  The user will be prompted to consent even if consent was granted before.
      */
-    CONSENT;
+    CONSENT,
+
+    /**
+     * acquireToken will not send the prompt parameter to the authorize endpoint.  The user may be prompted to login or to consent as required by the request.
+     */
+    WHEN_REQUIRED;
 
     @Override
     public String toString() {
         switch (this) {
             case SELECT_ACCOUNT:
                 return SELECT_ACCOUNT.name().toLowerCase();
-            case FORCE_LOGIN:
-                return "login";
+            case LOGIN:
+                return LOGIN.name().toLowerCase();
             case CONSENT:
                 return CONSENT.name().toLowerCase();
+            case WHEN_REQUIRED:
+                return WHEN_REQUIRED.name().toLowerCase();
             default:
                 throw new IllegalArgumentException();
         }
     }
 
     public OpenIdConnectPromptParameter toOpenIdConnectPromptParameter() {
+        String tag = Prompt.class.getSimpleName() + ":toOpenIdConnectPromptParameter";
         switch (this) {
             case SELECT_ACCOUNT:
                 return OpenIdConnectPromptParameter.SELECT_ACCOUNT;
-            case FORCE_LOGIN:
+            case LOGIN:
                 return OpenIdConnectPromptParameter.LOGIN;
             case CONSENT:
                 return OpenIdConnectPromptParameter.CONSENT;
+            case WHEN_REQUIRED:
+                String error = "WHEN_REQUIRED Does not have corresponding value in in the OIDC prompt enumeration.  It's meant to convey do not sent the prompt parameter.";
+                Logger.info(tag, error);
```

Comment:
Shouldn’t we return OpenIdConnectPromptParameter.None here ? If a developer passes in WhenRequired , wouldn’t this always throw an exception when converting to OpenIdConnectParameter for brokered auth ?

Reply:
WhenRequired means don't send a prompt parameter at all.